web/pdf_find_controller.js

```diff
@@ -771,13 +740,65 @@ class PDFFindController {
     return [isUnicode, query];
   }
 
-  #calculateMatch(pageIndex) {
-    let query = this.#query;
+  async #calculateMatch(pageIndex) {
+    const query = this.#query;
     if (query.length === 0) {
       return; // Do nothing: the matches should be wiped out already.
     }
-    const { caseSensitive, entireWord } = this.#state;
     const pageContent = this._pageContents[pageIndex];
+    const matcherResult = await this.match(query, pageContent, pageIndex);
```

Comment:
That method isn't async, so I really don't understand why `await` is necessary here!?

As far as I'm concerned this adds, completely undesirable, maintenance overhead since looking at the other code it's impossible to understand why this is done.